client/notebooks/Demo_Client_Notebook_Smartnoise-SQL.ipynb

```diff
@@ -800,415 +622,401 @@
-    "print(f\"Number of penguins: {nb_penguin}.\")\n",
-    "\n",
-    "std_bill_length = response['std_bill_length_mm'].iloc[0]\n",
-    "print(f\"Standard deviation of bill length: {np.round(std_bill_length, 2)}.\")"
+    "subquery = dp_query.source.relations[0].primary.query\n",
+    "syms = subquery._named_symbols\n",
+    "syms"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "367081be-1159-45d8-9129-88fba20fb697",
+   "id": "cc07d8c4-153f-4ad3-a977-a971b94d75aa",
    "metadata": {},
    "source": [
-    "She can now do all the postprocessing that she wants with the returned data without adding any privacy risk. "
+    "This last query with `group_by` will cost the same because `max_ids=1` (at most a penguin is once in the dataset) and so the `group_by` is applied on different partitions of the population."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 29,
-   "id": "f72b19d0",
+   "execution_count": 124,
+   "id": "5b69f3f2-07dd-48b8-9cd5-64eee53331f7",
    "metadata": {},
    "outputs": [
     {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "Standard error of bill length: 0.71.\n"
-     ]
+     "data": {
+      "text/plain": [
+       "{'epsilon_cost': 3.0, 'delta_cost': 5.000000000032756e-06}"
+      ]
+     },
+     "execution_count": 124,
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
-    "# Get standard error\n",
-    "standard_error = std_bill_length/np.sqrt(nb_penguin)\n",
-    "print(f\"Standard error of bill length: {np.round(standard_error, 2)}.\")"
+    "QUERY = \"SELECT COUNT(bill_length_mm) AS count_bl, AVG(bill_length_mm) AS avg_bl, STD(bill_length_mm) as std_bl FROM df GROUP BY species\"\n",
+    "cost = client.estimate_smartnoise_sql_cost(query = QUERY, epsilon = epsilon, delta = delta)\n",
+    "cost"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 30,
-   "id": "62630a03",
+   "cell_type": "markdown",
+   "id": "e20c4673-2c7b-44d5-bd7f-be88d6432a70",
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "The 95% confidence interval of the bill length of all penguins is [42.8, 45.57].\n"
-     ]
-    }
-   ],
    "source": [
-    " # Compute the 95% confidence interval\n",
-    "ZSCORE = 1.96\n",
-    "lower_bound, upper_bound = avg_bill_length - ZSCORE*standard_error, avg_bill_length + ZSCORE*standard_error\n",
-    "print(f\"The 95% confidence interval of the bill length of all penguins is [{np.round(lower_bound, 2)}, {np.round(upper_bound, 2)}].\")"
+    "NOTE: in the current code of Smartnoise-SQL, there is no odometer. Meaning all queries are independant. If someone first query the private dataset for a count, then a second time for the average and then for the standard deviation then the total cost will be added: 3 count + 2 average + 1 std. That's why it is better to do all in one query."
```

Comment:
If someone first querieS
That's why it is better to do everything in one query (or a single query).